mtp_cashbook/assets-src/javascripts/modules/moj.dialog.js

```diff
@@ -0,0 +1,107 @@
+// Dialog module
+// Dependencies: moj, _, jQuery
+
+/* globals console */
+(function () {
+  'use strict';
+
+  moj.Modules.Dialog = {
+    selector: '.js-Dialog',
+
+    init: function () {
+      _.bindAll(this, 'render', 'closeDialog', 'onKeyUp');
+      this.cacheEls();
+      this.bindEvents();
+    },
+
+    cacheEls: function () {
+      this.$body = $('body');
+      this.$backdrop = $('<div>').addClass('Dialog-backdrop');
+      this.$close = $('<a>');
+
+      // close element
+      this.$close
+        .attr('href', '#')
+        .attr('role', 'button')
+        .addClass('Dialog-close');
+    },
+
+    bindEvents: function () {
+      moj.Events.on('Dialog.render', this.render);
+      moj.Events.on('Dialog.close', this.closeDialog);
+      this.$body.on('click', this.selector, this.render);
+    },
+
+    render: function (e) {
+      this.$triggerEl = $(e.target);
+
+      e.preventDefault();
+      this.openDialog(this.$triggerEl.attr('href'));
+    },
+
+    openDialog: function (target) {
+      var $dialog = $(target);
+
+      // log warning if target doesn't exist
+      if ($dialog.length === 0) {
+        console.warn('The target element, ' + target + ', does not exist.');
```

Comment:
If you find it useful, keep it… but maybe appropriately wrap it in conditionals?